### DISCUSSION TITLE
可以更换一下代码的字体吗？

### DISCUSSION THREAD

**Discussion by lbn233** [ppoffice/hexo-theme-icarus#913] [Q&A]
现在的字体不知道是什么，就是博客默认的

但是 Fira Code 个人认为比现在的字体要好看

请问一下怎么更换代码字体，或者 developer 更换一下？

Thanks

**Comment by BlakeNeko** [comment]
遇到了类似的问题，就不单开一个discussion了。说说我的尝试过程。

首先尝试搜索了issue发现几个关于字体的讨论都是针对旧版本的主题，因此没能用上。

在阅读用户手册的时候发现默认的字体CDN就是谷歌，所以我就没有做修改而是直接在Google Fonts里选择了Noto Serif SC和Fira Code，复制了网页上提供的font-family中的字体名称，添加到 `node_modules\hexo-theme-icarus\include\style\base.styl` 中，生成博客文件后字体并没有发生变化，在开发者工具中能看到 `body` 那一块的CSS规则有显示自己添加的字体名，控制台也无报错，请问是我遗漏了什么步骤吗？

附上我修改的配置项，位于 node_modules\hexo-theme-icarus\include\style\base.styl`` 中

```
$family-sans-serif ?= 'Noto Serif SC', 'Microsoft YaHei', Ubuntu, Roboto, 'Open Sans', sans-serif
$family-code ?= 'Fira Code', 'Source Code Pro', monospace, 'Microsoft YaHei'
```

**Reply by ppoffice** [reply]
@BlakeNeko  你漏了这个:https://github.com/ppoffice/hexo-theme-icarus/blob/4.4.0/layout/common/head.jsx#L53-L56

**Reply by BlakeNeko** [reply]
好的，感谢提醒，已经修改成功。

**Reply by Doncic-77** [reply]
要怎么改呢？

**Comment by leimao** [comment]
@ppoffice 不懂就问。请问`family-sans-serif`和`family-code`分别是管网站里的什么部分的字体的？我看了半天源代码也没发现个所以然。另外，有什么比较优雅的方式可以分别设置导航栏，文章大标题，h1/h2/h3/... 等等的字体。比如说我导航栏字体想用A，文章大标题想用字体B，文章内部的小标题想用字体C，等等。谢谢。

**Reply by ppoffice** [reply]
`family-sans-serif`: 正文字体
`family-code`: 代码字体
没有办法通过设置单独修改每个元素的字体。想改只能在代码层面改。

**Reply by leimao** [reply]
感谢您的解答。另外再弱弱的问一下，“代码字体”中的“代码”是指文章里的coding block是吧？
